src/pfe/portal/routes/projects/links.route.js

```diff
@@ -0,0 +1,221 @@
+  try {
+    const { cw_user: user } = req;
+    const project = getProjectFromReq(req);
+    const { links } = project;
+
+    // If the link on the same PFE (local) fetch the projectURL from the ProjectList (ignore newProjectURL)
+    const { type: linkType, projectID: targetProjectID, projectURL: currentProjectURL } = links.get(currentEnvName);
+    
+    if (linkType === TYPES.LOCAL) {
+      // Get the url from the projectList, throw an error if the project does not exist
+      const localProjectURL = await verifyProjectExistsAndReturnInternalURL(user, targetProjectID);
+      newProjectURL = localProjectURL;
+    }
+
+    // If newEnvName or newProjectURL are not given through the API, default them to their old values
+    const updatedLinkEnvName = (newEnvName) ? newEnvName : currentEnvName;
+    const updatedLinkProjectURL = (newProjectURL) ? newProjectURL : currentProjectURL;
+    await project.updateLink(currentEnvName, updatedLinkEnvName, updatedLinkProjectURL);
+
+    // Send status and then kick off the restart/rebuild
+    res.sendStatus(204);
+
+    if (project.isOpen()) {
+      await restartOrBuildProject(user, project);
+    }
+  } catch (err) {
+    log.error(err);
+    if (err.code === ProjectLinkError.CODES.NOT_FOUND) {
+      res.sendStatus(404);
+    } else {
+      res.status(500).send(err);
+    }
+  }
+});
+
+router.delete('/api/v1/projects/:id/links', validateReq, checkProjectExists, async(req, res) => {
+  const envNameOfLinkToDelete = req.sanitizeBody('envName');
+  try {
+    const { cw_user: user } = req;
+    const project = getProjectFromReq(req);
+    await project.deleteLink(envNameOfLinkToDelete);
+
+    // Send status and then kick off the restart/rebuild
+    res.sendStatus(204);
+
+    if (project.isOpen()) {
+      await restartOrBuildProject(user, project);
+    }
+  } catch (err) {
+    log.error(err);
+    if (err.code === ProjectLinkError.CODES.NOT_FOUND) {
+      res.sendStatus(404);
+    } else {
+      res.status(500).send(err);
+    }
+  }
+});
+
+const verifyProjectExistsAndReturnInternalURL = async(user, projectID) => {
+  const project = await user.projectList.retrieveProject(projectID);
+  if (!project) {
+    throw new Error('projectID not found on local PFE');
+  }
+  const { host: dockerNetworkIP, ports: { internalPort }, appStatus } = project;
+  // If the project has not started return undefined as the projectURL
+  if (appStatus != 'started' && !dockerNetworkIP) {
+    return undefined;
+  }
+  // Attempt to get container name fall back to docker network IP address (given to us by file-watcher)
+  const container = await dockerInspect(project);
+  const host = (container && container.Name) ? container.Name.substring(1) : dockerNetworkIP;
+  return `${host}:${internalPort}`;
+}
+
+const restartOrBuildProject = async(user, project) => {
+  const { projectType } = project;
+  const projectTypesThatPickUpEnvsThroughRestart = ['nodejs', 'liberty', 'spring'];
```

Comment:
Is this going to work in Kubernetes? I think we have to rebuild node apps to restart them there.

Reply:
No at the moment this is Docker only

Reply:
That's fine - though should we do something like disable the links routes if PFE is running on Kubernetes for the moment?

Reply:
I can do that